stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/common/message/list/MessageListController.kt

```diff
@@ -1405,23 +1405,43 @@ public class MessageListController(
         ).enqueue { result ->
             if (result.isSuccess) {
                 val message = result.data()
+
                 message.attachments.removeAll { attachment ->
-                    if (attachmentToBeDeleted.assetUrl != null) {
-                        attachment.assetUrl == attachmentToBeDeleted.assetUrl
-                    } else {
-                        val isSame = attachment.imageUrl == attachmentToBeDeleted.imageUrl
-                        isSame
+                    val imageUrl = attachmentToBeDeleted.imageUrl
+                    val assetUrl = attachmentToBeDeleted.assetUrl
+
+                    when {
+                        assetUrl != null -> {
+                            attachment.assetUrl?.substringBefore("?") ==
+                                assetUrl.substringBefore("?")
+                        }
+                        imageUrl != null -> {
+                            attachment.imageUrl?.substringBefore("?") ==
+                                imageUrl.substringBefore("?")
+                        }
+                        else -> false
                     }
                 }
 
-                chatClient.updateMessage(message).enqueue(
-                    onError = { chatError ->
-                        logger.e {
-                            "Could not edit message to remove its attachments: ${chatError.message}. " +
-                                "Cause: ${chatError.cause?.message}"
+                if (message.text.isBlank() && message.attachments.isEmpty()) {
```

Comment:
Because we previously did not do this, we would leave behind an empty message if the last attachment was deleted.